src/openfermion/utils/_trotter_exp_to_qgates.py

```diff
@@ -220,8 +228,17 @@ def pauli_exp_to_qasm(qubit_operator_list,
             ret_list = ret_list + cnots1
 
             # 3. Rotation (Note kexp & Ntrot)
-            ret_list = ret_list + ["Rz {} {}".format(
-                term_coeff * evolution_time, qids[-1])]
+            if ancilla is not None:
+                if len(qids) > 0:
```

Comment:
Instead of checking qubit_list in a few places, it might be more compact to introduce one concept of it at the top of the function.  For example, if it's none, qubit_list is initialized to the ordered list of qubits that one would normally expect.  This should make the logic follow a bit more easily I think.

Reply:
This is a much better way of dealing with this, thanks!